### ISSUE TITLE
[Cannot read property 'classList' of null] on modal.hide() with footer

### ISSUE THREAD

**Issue by andrii-solokh** [trailheadapps/lwc-recipes#344]
### Summary

When closing modal with footer through @api hide() method from parent component, receiving error:
"[Cannot read property 'classList' of null]"

### Steps To Reproduce:

1. Create and deploy parent LWC.
2. In parent LWC add <c-modal> with footer slot:
```
<template>
    <c-modal header="Confirm deletion" class="deletion-modal">
        This action will remove the Products
        <div slot="footer" class="slds-align_absolute-center">
            <lightning-button label="Cancel" onclick={hideDeleteConfirmation}></lightning-button>
        </div>
    </c-modal>
</template>
```
3.  Handel button onclick event and hide modal:
```
hideDeleteConfirmation() {
        this.template.querySelector('.deletion-modal').hide()   
}
``` 

### Expected result

Expecting that modal will be hidden without errors

### Actual result

Getting "[Cannot read property 'classList' of null]" error.

### Additional information

![image](https://user-images.githubusercontent.com/10558860/88518833-cbba1e00-cff9-11ea-9b80-cb1ff4aff74b.png)

**Comment by pchittum** [comment]
Thanks for posting this @SAItm. 

It looks like your error is being surfaced from a function that you've not shown `handleSlotFooterChange`. 

I take it you're trying to take the code from the modal recipe and adapt it for use in a project. 

If you have a question about why your implementation is throwing the error, a better place to [post this question](https://salesforce.stackexchange.com/questions/ask) on the Salesforce StackExchange. 

You could post pretty much exactly your question above, but with a bit more context like: 

> I'm adapting the LWC Recipes modal recipe in a project and when doing so, getting this error on close of the modal window. 

Also be sure to pose the implementation of the`handleSlotFooterChange` function that's failing.

**Comment by andrii-solokh** [comment]
Thanks for response.
I didn't change handleSlotFooterChange, I left it as it is in lwc-receipies:
```
handleSlotFooterChange() {
        const footerEl = this.template.querySelector('footer');
        footerEl.classList.remove(CSS_CLASS);
}
```
After getting this error I just added:
```
handleSlotFooterChange() {
        const footerEl = this.template.querySelector('footer');
        if (footerEl) {
            footerEl.classList.remove(CSS_CLASS);
        }
    }
```

**Comment by pchittum** [comment]
Hmmm...the recipe, as is, currently works in my scratch org. And running debug in chrome tools it all seems good. I'd be curious to know more about your implementation and what else may have changed. Anything about environment, browser, etc. would be helpful.

**Comment by andrii-solokh** [comment]
@pchittum Sorry didn't have time to replay. Will update soon.

**Comment by andrii-solokh** [comment]
@pchittum I am back!
So what I've found is that after 'hide()' method is called, 'this.showModal' becomes false what triggers template rerender. Slot with name 'footer' changes (now it contains nothing), that invoke 'handleSlotFooterChange()' and if you will try to 
```
this.template.querySelector("footer");
```
That will return null. I have tested in my dev-ed org and it didn't show me error message, 'footerEl' variable is null in 'handleSlotFooterChange' method, when hide() was called. Try to catch it with break point.

Hope it will help.
